baldrick/github/github_api.py

```diff
@@ -494,27 +511,38 @@ def is_closed(self):
 class PullRequestHandler(IssueHandler):
 
     # https://developer.github.com/v3/checks/runs/#create-a-check-run
-    def set_check(self, name, summary, commit_hash='head', details_url=None,
-                  status='queued', conclusion='neutral'):
+    def set_check(self, external_id, title, name=None, summary=None, text=None,
+                  commit_hash='head', details_url=None, status=None,
+                  conclusion='neutral', check_id=None, completed_at=None):
         """
         Set check status.
 
         .. note:: This method does not provide API access to full
-                  check run capability (e.g., Markdown text, annotation,
+                  check run capability (e.g., annotation and
                   image). Add them as needed.
 
         Parameters
         ----------
-        name : str
-            Name of the check.
+        external_id : `str`
+            The reference for this check.
```

Comment:
Since you explained in the comment, might as well add it to docstring.
```suggestion
            The reference for this check.
            This is used by the ``towncrier`` plugin to provide a more user-friendly
            "name" for the check, while maintaining the ability to update it later.
```

Reply:
It isn't specific to towncrier, that's just the one I have implemented. external_id is designed for what we were using name for before.

Reply:
Feel free to rephrase my suggestion.